Pr branch

rashmi-kulkarni-ct402 opened the pull request.

rashmi-kulkarni-ct402 committed:
Initial commit in new branch (pr_branch)

rashmi-kulkarni-ct402 committed:
Updated README.md

rashmi-kulkarni-ct402 committed:
Initial commit for .obj to .txt translator

rashmi-kulkarni-ct402 committed:
Delete headers directory

rashmi-kulkarni-ct402 committed:
Delete src directory

rashmi-kulkarni-ct402 committed:
Delete main.cpp

rashmi-kulkarni-ct402 committed:
initial commit for .stl to .txt translator

rashmi-kulkarni-ct402 committed:
initial commit for .obj to .stl translator

rashmi-kulkarni-ct402 committed:
Update ObjReader.cpp

removed unused variable "vertexindex" from ObjReader.cpp file

rashmi-kulkarni-ct402 committed:
Update and rename ObjWriter.cpp to TxtWriter.cpp

rashmi-kulkarni-ct402 committed:
Update and rename ObjWriter.h to TxtWriter.h

rashmi-kulkarni-ct402 committed:
Update and rename ObjWriter.cpp to TxtWriter.cpp

rashmi-kulkarni-ct402 committed:
renamed ObjWriter to TxtWriter and objWriter to txtWriter

rashmi-kulkarni-ct402 committed:
Update and rename StlWriter.h to TxtWriter.h

rashmi-kulkarni-ct402 committed:
Update and rename StlWriter.cpp to TxtWriter.cpp

rashmi-kulkarni-ct402 committed:
Updated and renamed StlWriter to TxtWriter and stlWriter to txtWriter

rashmi-kulkarni-ct402 committed:
updated Triangulation.h and Triangulation.cpp files

Removed "std::vector<Point3D> &uniqueNormals()" and kts implementation

rashmi-kulkarni-ct402 committed:
Update Triangulation.h

removed "std::vector<Point3D> mNormals;"

rashmi-kulkarni-ct402 committed:
Update Triangulation.cpp

removed implementation -
std::vector<Shapes3D::Point3D> &Shapes3D::Triangulation::normals()
{
    return mNormals;
}

rashmi-kulkarni-ct402 committed:
Delete Obj_To_Txt/Triangulation.cpp

rashmi-kulkarni-ct402 committed:
Delete Obj_To_Txt/Triangulation.h

rashmi-kulkarni-ct402 committed:
updated Point.h, Triangle.h, Triangulation.h

rashmi-kulkarni-ct402 committed:
Update Triangulation.cpp

removed implementation of vector for normal

rashmi-kulkarni-ct402 committed:
initial commit to Stl_To_Obj

rashmi-kulkarni-ct402 committed:
Update TxtWriter.cpp

rashmi-kulkarni-ct402 committed:
Update main.cpp

adeshlad2 commented on Obj_To_Stl/headers/Triangle.h:
```diff
@@ -0,0 +1,28 @@
+        void setindex3(int inIndex3);
+
+    private:
+        int mIndex1;
```
Triangle should always store 3 Point3D vertices and 1 Point3D normal.

rashmi-kulkarni-ct402 replied:
Triangle class has "int" variables (we are declaring indices and so it is taken as "int") as mentioned by Vinayak. I have later mapped "Point3D" and with "int" indices in Reader file.

adeshlad2 commented on Obj_To_Stl/headers/Triangle.h:
```diff
@@ -0,0 +1,28 @@
+    {
+    public:
+        Triangle();
+        Triangle(int inIndex1, int inIndex2, int inIndex3, int inNormalIndex);
```
change methods for Point3D compatibility.

rashmi-kulkarni-ct402 replied:
Triangle class has "int" variables (we are declaring indices and so it is taken as "int") as mentioned by Vinayak. I have later mapped "Point3D" and with "int" indices in Reader file.

adeshlad2 commented on Obj_To_Stl/headers/Triangulation.h:
```diff
@@ -0,0 +1,23 @@
+
+    private:
+        std::vector<Point3D> mUniquePoints;
+        std::vector<Triangle> mTriangles;
```
Triangulation should only store Triangle-vector. Unique points can be stored in different class.

adeshlad2 commented on Obj_To_Stl/headers/Triangulation.h:
```diff
@@ -0,0 +1,23 @@
+        ~Triangulation();
+
+        std::vector<Point3D> &uniquePoints();
+        std::vector<Triangle> &triangles();
```
Returning reference is not recommended as mTriangles is set private. make methods to add and remove element from mTriangles.

rashmi-kulkarni-ct402 replied:
updated this by adding functions to add point, triangles into traingulation object instead of using references

adeshlad2 commented on Obj_To_Stl/headers/Triangulation.h:
```diff
@@ -0,0 +1,23 @@
+        Triangulation();
+        ~Triangulation();
+
+        std::vector<Point3D> &uniquePoints();
```
Remove other functions. Add to different class.

adeshlad2 commented on Obj_To_Stl/src/ObjReader.cpp:
```diff
@@ -0,0 +1,93 @@
+        {
+            double x, y, z;
+            vertexStream >> x >> y >> z;
+            Point3D point(x, y, z);
```
store this point into points vector in this function, no need for new class, as new points will always be unique.

rashmi-kulkarni-ct402 replied:
As we require Point3D (points) and int (indices) for mapping, these both are included in Triangulation class. The same goes with Point3D (normal) and int (indices).

adeshlad2 commented on Obj_To_Stl/src/ObjReader.cpp:
```diff
@@ -0,0 +1,93 @@
+        {
+            double x, y, z;
+            vertexStream >> x >> y >> z;
+            Point3D normal(x, y, z);
```
same as point

rashmi-kulkarni-ct402 replied:
As we require Point3D (points) and int (indices) for mapping, these both are included in Triangulation class. The same goes with Point3D (normal) and int (indices).

adeshlad2 reviewed (changes requested):
Comments are added to single project, required changes to be made in all projects. Will check writer later.

rashmi-kulkarni-ct402 committed:
Update ObjReader.cpp

rashmi-kulkarni-ct402 committed:
removed inNormalIndex from Triangle constructor

rashmi-kulkarni-ct402 committed:
removed inNormalIndex from Triangle constructor

rashmi-kulkarni-ct402 committed:
Update Triangle.h

added NormalIndex setter

rashmi-kulkarni-ct402 committed:
Update Triangle.cpp

added normalIndex setter

rashmi-kulkarni-ct402 committed:
Update StlReader.cpp

rashmi-kulkarni-ct402 committed:
Update ObjReader.cpp

removed unused vectors 
1. std::vector<Point3D> points;
 2. std::vector<Triangle> triangles;
 3. std::vector<Point3D> normals;

rashmi-kulkarni-ct402 committed:
added 3 function declarations

void addUniquePointToTriangulation(const Point3D point);
void addTriangleToTriangulation(const Triangle triangle);
void addUniqueNormalToTriangulation(const Point3D normal);

rashmi-kulkarni-ct402 committed:
Add files via upload

Triangulation.cpp --> 
1. removed references to vectors
2. added function definitions for 
void addUniquePointToTriangulation(const Point3D point);
void addTriangleToTriangulation(const Triangle triangle);
void addUniqueNormalToTriangulation(const Point3D normal);

ObjReader.cpp --> 1. added triangles by calling addUniqueNormalToTriangulation() function
2. Added comments

StlWriter.cpp --> 1. removed references to vectors
2. Added comments

rashmi-kulkarni-ct402 committed:
Update ObjReader.cpp

rashmi-kulkarni-ct402 committed:
update to Triangulation.h

Triangulation.h -->
added 2 function declarations for
void addUniquePointToTriangulation(const Point3D point);
void addTriangleToTriangulation(const Triangle triangle);

rashmi-kulkarni-ct402 committed:
update to Triangulation.cpp, ObjReader.cpp & TxtWriter.cpp 

Triangulation.cpp --> 
1. removed references to vectors
2. added function definitions for 
void addUniquePointToTriangulation(const Point3D point);
void addTriangleToTriangulation(const Triangle triangle);

ObjReader.cpp --> 1. added triangles by calling addTriangleToTriangulation() function
2. Added comments

TxtWriter.cpp --> 1. removed references to vectors
2. Added comments

rashmi-kulkarni-ct402 committed:
update to Triangulation.h

Triangulation.h --> 
1. added function declarations for
void addUniquePointToTriangulation(const Point3D point);
void addTriangleToTriangulation(const Triangle triangle);
void addUniqueNormalToTriangulation(const Point3D normal);

rashmi-kulkarni-ct402 committed:
update to Triangulation.cpp, StlReader.cpp & ObjWriter.cpp 

Triangulation.cpp --> 
1. removed references to vectors
2. added function definitions for 
void addUniquePointToTriangulation(const Point3D point);
void addTriangleToTriangulation(const Triangle triangle);
void addUniqueNormalToTriangulation(const Point3D normal);

StlReader.cpp --> 1. added triangles by calling addTriangleToTriangulation() function
2. Added comments

TxtWriter.cpp --> 1. removed references to vectors
2. Added comments

rashmi-kulkarni-ct402 committed:
update to Triangulation.h

Triangulation.h --> 
1. added function declarations for
void addUniquePointToTriangulation(const Point3D point);
void addTriangleToTriangulation(const Triangle triangle);

rashmi-kulkarni-ct402 committed:
updated inParameter names

rashmi-kulkarni-ct402 committed:
Updated inParameter names

rashmi-kulkarni-ct402 committed:
Updated inParameter names

rashmi-kulkarni-ct402 committed:
update to Triangulation.cpp, StlReader.cpp & TxtWriter.cpp 	

Triangulation.cpp --> 
1. removed references to vectors
2. added function definitions for 
void addUniquePointToTriangulation(const Point3D point);
void addTriangleToTriangulation(const Triangle triangle);

StlReader.cpp --> 1. added triangles by calling addTriangleToTriangulation() function
2. Added comments

TxtWriter.cpp --> 1. removed references to vectors
2. Added comments
